apisix/runner/server/server.py

```diff
@@ -84,6 +84,12 @@ def __init__(self, config: NewServerConfig):
         self.sock.bind(self.fd)
         self.sock.listen(1024)
 
+        # the default socket permission is 0755, which prevents the 'nobody' worker process
+        # from writing to it if the APISIX is run under root.
+        os.chmod(self.fd, 0o766)
+        if os.stat(self.fd).st_mode & 0xfff != 0o766:
+            raise Exception("can't change mode for unix socket permission to 766")
```

Comment:
Could you update the CI configuration so the e2e test can pass?
We need to adjust the port of Admin API.